### PR TITLE
Mac: Add support for EnergyScan operation at mac layer.

### PR DESCRIPTION
**Pull request by abtink**
This commit adds support for energy scan operation at mac layer.

### CODE REVIEW

**Comment by abtink** [comment]
@jwhui I set this PR for code review now. I'd like to add the counterpart for energy-scan in wpantund and test everything together. Can you please merge this after the test? Thanks.

File: src/core/mac/mac.cpp [hunk 1]
```diff
@@ -131,12 +134,26 @@ Mac::Mac(ThreadNetif &aThreadNetif):
 
 ThreadError Mac::ActiveScan(uint32_t aScanChannels, uint16_t aScanDuration, ActiveScanHandler aHandler, void *aContext)
 {
+    mActiveScanHandler = aHandler;
```
Comment (jwhui) [inline src/core/mac/mac.cpp]: Set this only if `Scan()` succeeds?

File: src/core/mac/mac.cpp [hunk 2]
```diff
@@ -131,12 +134,26 @@ Mac::Mac(ThreadNetif &aThreadNetif):
...
+
+    return Scan(kScanTypeActive, aScanChannels, aScanDuration, aContext);
+}
+
+ThreadError Mac::EnergyScan(uint32_t aScanChannels, uint16_t aScanDuration, EnergyScanHandler aHandler, void *aContext)
+{
+    mEnergyScanHandler = aHandler;
```
Comment (jwhui) [inline src/core/mac/mac.cpp]: Set this only if `Scan()` succeeds?

Comment (abtink) [inline src/core/mac/mac.cpp]: Yes. indeed. I will fix it.

**Comment by abtink** [comment]
The counterpart of this in wpantund is in [PR55](https://github.com/openthread/wpantund/pull/55).
Both changes are tested/verified together. This is ready for review and merge. Thanks

**Comment by turon** [comment]
LGTM :+1:

**Pull request merged**